src/main/java/com/minecolonies/coremod/commands/colonycommands/ShowColonyInfoCommand.java

```diff
@@ -56,11 +70,70 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
         return super.getCommandUsage(sender) + "<ColonyId|OwnerName>";
     }
 
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu) throws CommandException
+    {
+        // See if we have a valid colony,
+        Colony colony = actionMenu.getColonyForArgument("colony");
+        EntityPlayer player = null;
+        if (null == colony)
+        {
+            // see if we have a valid player
+            player = actionMenu.getPlayerForArgument("player");
+            if (null != player)
+            {
+                final IColony iColony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), player);
+                if (iColony != null)
+                {
+                    if (!canPlayerUseCommand(player, SHOWCOLONYINFO, iColony.getID()))
+                    {
+                        sender.sendMessage(new TextComponentString(NOT_PERMITTED));
+                        return;
+                    }
+                    colony = ColonyManager.getColony(iColony.getID());
+                }
+            }
+        }
+
+        if (null == colony)
+        {
+            // see if we have a sender that is a valid player
+            if (sender instanceof EntityPlayer)
+            {
+                player = (EntityPlayer) sender;
+                final UUID mayorID = player.getUniqueID();
+                final IColony iColony = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID);
+                if (iColony != null)
+                {
+                    if (!canPlayerUseCommand(player, SHOWCOLONYINFO, iColony.getID()))
+                    {
+                        sender.sendMessage(new TextComponentString(NOT_PERMITTED));
+                        return;
+                    }
+                    colony = ColonyManager.getColony(iColony.getID());
+                }
+            }
+        }
+
+        if (colony == null)
+        {
+            if (null != player)
+            {
+                sender.sendMessage(new TextComponentString(String.format(NO_COLONY_FOR_PLAYER_FOUND_MESSAGE, player.getName())));
+            }
+            else
+            {
+                sender.sendMessage(new TextComponentString(String.format(NO_COLONY_FOUND_MESSAGE)));
+            }
+            return;
+        }
+
+        executeShared(server, sender, colony);
+    }
+
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)